pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -575,6 +602,9 @@ func (p *PriorityQueue) SchedulingCycle() int64 {
 func (p *PriorityQueue) AddUnschedulableIfNotPresent(logger klog.Logger, pInfo *framework.QueuedPodInfo, podSchedulingCycle int64) error {
```

Comment:
It seems wrong that we pass a `podSchedulingCycle` instead of obtaining it inside this function.
This could be causing data races, as we obtain the lock twice.

Could you fix that in a separate PR before this?

Alternatively, should we have been using the schedulingCycle at the time of poping (the value that you are storing now in `inFlightPod`)?